src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SocketConnection.cs

```diff
@@ -316,33 +340,68 @@ private void CloseAsyncAndLinger()
 
             try
             {
-                // A FIN (shutdown) packet has already been sent to the remote host and we're waiting for the remote
-                // host to send a FIN back. A pending read on a socket will complete returning zero bytes when a FIN
```

Comment:
This comment is worth keeping as it describes what the code is doing here.

Reply:
👍